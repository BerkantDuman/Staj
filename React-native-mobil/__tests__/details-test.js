/**
 * @format
 */

import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Details from '../src/screens/details'
import Data from '../src/screens/home';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


test('Api test case', async function () {
    global.fetch = jest.fn().mockImplementation(() => {
        var p = new Promise((resolve, reject) => {
            resolve({
                json: function () {
                    return { id: 2 }
                }
            })
        })
        return p;
    })
    const response = await Data.all();
    console.warn(response);
    expect(response.id).toBe(2);
})

test('renders correctly', () => {
    const navigationMock = { state: { params: { id: 1 } } };
    const tree = renderer.create(<Details navigation={navigationMock} />).toJSON();
    expect(tree).toMatchSnapshot();
});
