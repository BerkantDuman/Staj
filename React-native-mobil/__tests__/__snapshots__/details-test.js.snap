// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "pink",
    }
  }
>
  <View>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Firstname"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 8,
          "color": "black",
          "marginHorizontal": 10,
          "marginVertical": 15,
        }
      }
      underlineColorAndroid="transparent"
      value={null}
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Lastname"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 8,
          "color": "black",
          "marginHorizontal": 10,
          "marginVertical": 15,
        }
      }
      underlineColorAndroid="transparent"
      value={null}
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Address"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 8,
          "color": "black",
          "marginHorizontal": 10,
          "marginVertical": 15,
        }
      }
      underlineColorAndroid="transparent"
      value={null}
    />
    <View
      style={
        Object {
          "borderRadius": 8,
          "marginHorizontal": 10,
          "marginVertical": 15,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Update
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
